### PR TITLE
feat: require compile-time flag to enable liquidator query

### PR DESCRIPTION
**Pull request by toteki**
## Description

This prevents nodes not specifically built for liquidators from running the inefficient `liquidation_targets` query.

closes: #1315 

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [x] added appropriate labels to the PR
- [x] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] provided a link to the relevant issue or specification
- [x] added a changelog entry to `CHANGELOG.md`
- [x] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/leverage/keeper/grpc_query.go [hunk 1]
```diff
@@ -10,6 +11,9 @@ import (
 	"github.com/umee-network/umee/v3/x/leverage/types"
 )
 
+// EnableLiquidator must be true at compile time to enable QueryLiquidationTargets
```
Comment (toteki) [inline x/leverage/keeper/grpc_query.go:14]:
```suggestion
// EnableLiquidator must be set to "true" at compile time to enable QueryLiquidationTargets
```

File: Makefile [hunk 2]
```diff
@@ -71,6 +73,7 @@ BUILD_FLAGS := -tags "$(build_tags)" -ldflags '$(ldflags)'
 
 build: go.sum
 	@echo "--> Building..."
+	@echo "LIQUIDATOR:" $(LIQUIDATOR)
```
Comment (RafilxTenfen) [inline Makefile:76]:
```suggestion
```

File: Makefile [hunk 3]
```diff
@@ -80,8 +83,12 @@ build-no_cgo:
 build-linux: go.sum
 	LEDGER_ENABLED=false GOOS=linux GOARCH=amd64 $(MAKE) build
 
+build-liquidator:
+	LIQUIDATOR=true $(MAKE) build
+
 install: go.sum
 	@echo "--> Installing..."
+	@echo "LIQUIDATOR:" $(LIQUIDATOR)
```
Comment (RafilxTenfen) [inline Makefile:91]:
```suggestion
```

**Review by RafilxTenfen — APPROVED** [review]
I don't think we need to tell the value of `LIQUIDATOR` explicitly at `make build` and `make install`

**Comment by robert-zaremba** [comment]
Documentation is missing: both in the spec directory and release notes.

**Pull request merged**